src/utils/daos.c

```diff
@@ -1071,8 +1071,63 @@ print_oclass_names_list(FILE *stream)
 	free(str);
 }
 
+#define FIRST_LEVEL_HELP() \
+do { \
+	fprintf(stream, \
+	"usage: daos RESOURCE COMMAND [OPTIONS]\n" \
+	"resources:\n" \
+	"	  pool             pool\n" \
+	"	  container (cont) container\n" \
+	"	  object (obj)     object\n" \
+	"	  version          print command version\n" \
+	"	  help             print this message and exit\n"); \
+	fprintf(stream, "\n"); \
+	fprintf(stream, "use 'daos help RESOURCE' for resource specifics\n"); \
+} while (0)
+
+#define ALL_CONT_CMDS_HELP() \
+do { \
+	fprintf(stream, "\n" \
+	"container (cont) commands:\n" \
+	"	  create           create a container\n" \
+	"	  destroy          destroy a container\n" \
+	"	  list-objects     list all objects in container\n" \
+	"	  list-obj\n" \
+	"	  query            query a container\n" \
+	"	  get-prop         get all container's properties\n" \
+	"	  set-prop         set container's properties\n" \
+	"	  get-acl          get a container's ACL\n" \
+	"	  overwrite-acl    replace a container's ACL\n" \
+	"	  update-acl       add/modify entries in a container's ACL\n" \
+	"	  delete-acl       delete an entry from a container's ACL\n" \
+	"	  set-owner        change the user and/or group that own a container\n" \
+	"	  stat             get container statistics\n" \
+	"	  list-attrs       list container user-defined attributes\n" \
+	"	  del-attr         delete container user-defined attribute\n" \
+	"	  get-attr         get container user-defined attribute\n" \
+	"	  set-attr         set container user-defined attribute\n" \
+	"	  create-snap      create container snapshot (optional name)\n" \
+	"			   at most recent committed epoch\n" \
+	"	  list-snaps       list container snapshots taken\n" \
+	"	  destroy-snap     destroy container snapshots\n" \
+	"			   by name, epoch or range\n" \
+	"	  rollback         roll back container to specified snapshot\n"); \
+	fprintf(stream, "\n"); \
+	fprintf(stream, "use 'daos help cont|container COMMAND' for command specific options\n"); \
+} while (0)
+
+#define ALL_BUT_CONT_CREATE_OPTS_HELP() \
```

Comment:
(style)  Single statement macros should not use a do {} while (0) loop

Reply:
Needn't be enclosed since it's just an fprintf. Looks like initially it was just complaining about the semicolon.

Reply:
Yes I found this too .... (looks like checkpatch is never happy with my code!), but I try to finish analyse the CI failures prior to push a new (last?) commit/version.